### DISCUSSION TITLE
Making Discord Notifications Look Better

### DISCUSSION THREAD

**Discussion by rohitp30** [upptime/upptime#206] [Ideas]
Is there a way to modify how Upptime sends a notification to Discord? Right now the message it sends is quite weird and I was wondering if there was a way to change how it appeared. 

[Webhook Right Now](https://gyazo.com/4ff600c86ca87a0897bb01545f26da69)

**Comment by Soneji** [comment]
What would you prefer it to look like? Please give some ideas :)

**Reply by rohitp30** [reply]
Before switching to Upptime, I created another bot to track my main discord bot's presence. Every time it was marked offline it would send a message like this: 

It doesn't have to be exactly like this but like something that can follow this format in an embed:
[Past Embed](https://gyazo.com/9fb894a26701685d4d4cf976b91ba3c8)

**Comment by Soneji** [comment]
This looks like it can be easily implemented by changing 

https://github.com/upptime/uptime-monitor/blob/a1aa9a442b1a9a0e33e9c0445e1cf46d96f33fb6/src/helpers/notifme.ts#L224

to an embed type instead of content

The embed type is documented well here: https://birdie0.github.io/discord-webhooks-guide/structure/embeds.html

**Reply by Soneji** [reply]
I was wondering if there is a way in notifme.ts to check whether it is a down message or up message? Then we can change the colour appropriately

**Reply by forerosantiago** [reply]
Why not making configurations in `.upptimerc.yml` like the intro message and intro title?

**Reply by rohitp30** [reply]
I don't think you can really modify it using a PR, looks like it's gettings its content by an environment variable. 

```
import { replaceEnvironmentVariables } from "./environment";

message = replaceEnvironmentVariables(message);
```

**Reply by Soneji** [reply]
I dont understand what you are referring to @Space-Turtle0

**Reply by rohitp30** [reply]
The link you directed here
(https://github.com/upptime/uptime-monitor/blob/a1aa9a442b1a9a0e33e9c0445e1cf46d96f33fb6/src/helpers/notifme.ts#L224)

The `content` is being defined/directed towards an environment, we can't really access those or modify those in a PR.

**Reply by Soneji** [reply]
we can still change 
```js
content: message
```
to 
```js
"embeds": [{
    "title": "Upptime!" // or say for example repo name/site name
    "description": message
  }]
```

right??

**Reply by rohitp30** [reply]
True, but the `description` itself could still use some modifications.

**Reply by rohitp30** [reply]
Looks like we might need someone else who has access to the environments to help us out heh.

**Reply by Soneji** [reply]
The replace environment variables function is simply this file: 

https://github.com/upptime/uptime-monitor/blob/master/src/helpers/environment.ts

which I think just replaces any variables with the ones from GitHub action secrets

**Reply by AnandChowdhary** [reply]
Can someone give me a template for what you'd like the embed to look like? I'd be happy to make the change!

**Reply by rohitp30** [reply]
@AnandChowdhary Yup! I'll draft one up!

**Reply by rohitp30** [reply]
@AnandChowdhary 
Down Webhook: 
https://share.discohook.app/go/lzqgj18v

Up Webhook:
https://share.discohook.app/go/0m4fp8ej

You can view the JSON Editor at the very bottom of the embed editor.

**Reply by AnandChowdhary** [reply]
@Space-Turtle0, I'm getting a 404 here. 😢 Can you share it again or using a different service?

**Reply by ririko5834** [reply]
It was removed, why you just dont make custom notifications support, that everyone can setup everything that want?

**Reply by rohitp30** [reply]
Sorry for the late response @AnandChowdhary 

Here are some screenshots
UP: https://gyazo.com/993792bdb727a4eaee15bd9c2bad930f
DOWN: https://gyazo.com/c3c31dfbef08fe591d9b002a9c624f9f

Everything in quotes (Service, timestamp, URL HERE) should basically be replaced with the user has configured!

**Reply by ririko5834** [reply]
Please add also option to build your own embed.

**Reply by FxllenCode** [reply]
@Space-Turtle0, @AnandChowdhary:

I use discord integration with my project a lot, it currently looks really terrible. The link you sent isn't working anymore, but the screenshots look really good! I think improving discord notifications could be a great QoL feature for upptime.

**Reply by AnandChowdhary** [reply]
Happy to accept a PR @FxllenCode

**Comment by ririko5834** [comment]
Please add option to setup own request url, method, body and headers in .env to make custom notication.
for example you can send Discord Embed using Discord webhook

**Comment by ririko5834** [comment]
Discord embed notifications will look better than text one. Add option to setup own embed or add custom notifications, where you can setup own request url, method, body and headers in .env

**Comment by ririko5834** [comment]
Please dont ignore this https://github.com/upptime/upptime/discussions/191
**If this will be added, you can make for example custom Discord embed notifications**

**Comment by ririko5834** [comment]
When this will be added?

**Comment by wolfier74** [comment]
I like this idea. Embeds looks better than plain messages.

**Comment by LukePrior** [comment]
This sounds like a good idea.
